### DISCUSSION TITLE
RNS β 0.3.8

### DISCUSSION THREAD

**Discussion by markqvist** [markqvist/Reticulum#68] [Announcements]
This release brings big improvements to compatibility with various system types, along with several convenient new features, and a lot of tuning, optimisation and stability improvements. In a continued effort, the documentation has also been updated, restructured, and had several new and informative sections added.

**Changes**
- Added ability to install and run RNS without any dependencies
- Added backend abstraction for cryptographic primitives
- Added pure-python implementations of all cryptographic primitives
- Added accept option to Link API
- Added several undocumented API calls to the documentation
- Added option to filter interfaces to rnstatus utility
- Added "Communications Hardware" chapter to the documentation
- Improved multiple chapters and restructured documentation
- Improved efficiency of Transport instances
- Improved performance of Resource transfers
- Improved Resource handling strategies over different physical link types
- Improved link capacity and speed estimation calculations
- Improved I2P interface error handling and stability
- Tuned Resource and Link timeouts
- Tuned TCP socket options for better reliability over intermittent links
- Tuned I2P interface timeouts for better reliability over intermittent links
- Fixed a missing check for zero-length packets on IFAC-enabled interfaces
- Fixed a socket allocation leak in I2P interfaces
- Added unit tests
- Added performance profiling tools
- Improved build system

<hr /><em>This discussion was created from the release <a href='https://github.com/markqvist/Reticulum/releases/tag/0.3.8'>RNS β 0.3.8</a>.</em>

**Comment by markqvist** [comment]
I am really happy to finally have this one released. It marks an important milestone for me, since Reticulum can now run on *any* system that supports Python3, making it accessible even on all kinds of obscure system types.

I *think* I also finally managed to get the I2P interfaces to a completely reliable state, where they will survive all kinds of abuse, like the I2P daemon completely disappearing or crashing, and then reappearing later.

More news on RNodes and other interesting stuff soon, as I promised a while ago :)

**Reply by timp34** [reply]
have just updated and restarted and already seen the I2p connections re-establishing much quicker esp if I restart i2p router.
Great work! Looking forward to testing out the other updates. :)
